okhttp/src/main/java/okhttp3/CacheControl.kt

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright (C) 2019 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3
+
+import okhttp3.internal.http.HttpHeaders
+import java.util.concurrent.TimeUnit
+
+/**
+ * A Cache-Control header with cache directives from a server or client. These directives set policy
+ * on what responses can be stored, and which requests can be satisfied by those stored responses.
+ *
+ * See [RFC 7234, 5.2](https://tools.ietf.org/html/rfc7234#section-5.2).
+ */
+class CacheControl private constructor(
+  /**
+   * In a response, this field's name "no-cache" is misleading. It doesn't prevent us from caching
+   * the response; it only means we have to validate the response with the origin server before
+   * returning it. We can do this with a conditional GET.
+   *
+   * In a request, it means do not use a cache to satisfy the request.
+   */
+  @get:JvmName("noCache") val noCache: Boolean,
```

Comment:
Isn't this breaking your rule of compatibility? This is a source breaking change for Kotlin users. Or did we relax that?

Reply:
Oooh, it does. Lemme fix.

Reply:
I've tried to revert almost all test changes by the end to ensure the code is at least source compatible.  But I understood we really want binary compatible via the build checks.